include/reactor-uc/action.h

```diff
@@ -10,55 +10,21 @@ typedef struct LogicalAction LogicalAction;
 typedef struct PhysicalAction PhysicalAction;
 
 struct Action {
-  Trigger super;          // Inherit from Trigger
-  interval_t min_offset;  // The minimum offset from the current time that an event can be scheduled on this action.
-  interval_t min_spacing; // The minimum spacing between two consecutive events on this action.
-  tag_t previous_event;   // Used to enforce min_spacing
-  TriggerEffects effects; // The reactions triggered by this Action.
-  TriggerSources sources; // The reactions that can write to this Action.
-  TriggerDataQueue trigger_data_queue; // FIFO storage of the data associated with the events scheduled on this action.
+  Trigger super;         // Inherit from Trigger
+  interval_t min_offset; // The minimum offset from the current time that an event can be scheduled on this action.
+  bool is_physical;
```

Comment:
I personally prefer enums for this for better code readability in cases like:

```c
if (action.type == ACTION_LOGICAL) {}
```

vs.

```c
if (!action.is_physical()) {}
```

But I don't have a strong opinion on this. I also think it is okay as is.